src/main/java/io/curity/identityserver/plugin/linkedin/authentication/CallbackRequestHandler.java

```diff
@@ -145,12 +147,14 @@ public CallbackGetRequestModel preProcess(Request request, Response response)
 
     private Map<String, Object> redeemCodeForTokens(CallbackGetRequestModel requestModel)
     {
+        HttpRequest.BodyProcessor body = getFormEncodedBodyFrom(createPostData(_config.getClientId(),
+                _config.getClientSecret(),
+                requestModel.getCode(), createRedirectUri(_authenticatorInformationProvider, _exceptionFactory)));
         HttpResponse tokenResponse = getWebServiceClient("www.linkedin.com")
                 .withPath("/oauth/v2/accessToken")
                 .request()
                 .contentType("application/x-www-form-urlencoded")
-                .body(getFormEncodedBodyFrom(createPostData(_config.getClientId(), _config.getClientSecret(),
-                        requestModel.getCode(), requestModel.getRequestUrl())))
```

Comment:
this was the bug. It differed from the redirect_uri sent to the LinkedIn authorization endpoint